enterprise/server/cmd/ci_runner/main.go

```diff
@@ -2526,11 +2626,23 @@ func runBazelWrapper() error {
 	if err != nil {
 		return err
 	}
-	args = append([]string{bazelCmd}, append(startupArgs, args...)...)
 
-	// Replace the process running the bazel wrapper with the process running bazel,
-	// so there are no remaining traces of the wrapper script.
-	return syscall.Exec(bazelCmd, args, os.Environ())
+	iid, err := newUUID()
+	if err != nil {
+		return err
+	}
+	args = appendBazelSubcommandArgs(args, fmt.Sprintf("--invocation_id=%s", iid))
+	args = append(startupArgs, args...)
+
+	io.WriteString(os.Stdout, fmt.Sprintf("BB command for invocation [%s]: [%s]", iid, strings.Join(os.Args[1:], " ")))
+	start := time.Now()
+	err = runCommand(context.Background(), bazelCmd, args, nil, "", os.Stderr)
```

Comment:
@bduffany You've previously mentioned how we should use `exec` to avoid having child processes. I couldn't think of a better way to capture the exit code of the bazel commands though. Let me know if you have any better ideas

Another way would be to stop including this information in the BES events and have the UI fetch each child invocation to get the status, but that felt like a poor solution because we may have to fetch the invocations thousands of times for long builds

Reply:
hm I wonder if we could simplify by storing the parent workflow invocation ID in the invocation table, and then when we call LookupInvocation for displaying the invocation in the UI, we also fetch child invocations and return their metadata back to the UI?

I think we already set the necessary metadata on the child invocation (since we render the parent workflow ID in the breadcrumbs) but just aren't storing it in the DB

(tbh I'm not sure why I didn't consider this option before... are there any downsides you can think of?)

Reply:
oh, I think that's what you meant by

> Another way would be to stop including this information in the BES events and have the UI fetch each child invocation to get the status, but that felt like a poor solution because we may have to fetch the invocations thousands of times for long builds

I think it's fine as long as we're only fetching the child invocations' DB row, and not their full BES stream

Reply:
Sounds good - this will also fix a confusing UI experience where sometimes from the outer invocation, the duration and status of the builds is incorrect, because there are 2 sources of truth (outer invocation uses BES event, inner invocation uses invocation DB row)

Reply:
@bduffany Cleaned up this PR after refactoring the FE to fetch the child invocations directly from the db